### PR TITLE
CE-432 Added SEO check to Word Count

### PR DESCRIPTION
**Pull request by vpratti**

### CODE REVIEW

File: plugins/wordcount/plugin.js [hunk 1]
```diff
@@ -188,6 +191,37 @@ CKEDITOR.plugins.add("wordcount", {
...
+            toggleSEO(editorInstance, wordCount);
+        }
+
+        function getTextBeforeFirstSEOObject(text) {
+            var assetCardSeperator = '<div asset="::asset"';
+            var relatedLinkSeperator = '<span class="exclude-from-newsgate';
+            var indexOfAssetCard = text.indexOf(assetCardSeperator);
+            var indexOfRelatedLink = text.indexOf(relatedLinkSeperator);
+
+            if (indexOfRelatedLink != -1 && (indexOfAssetCard == -1 || indexOfRelatedLink < indexOfAssetCard)) {
```
Comment (RyanCopley) [inline plugins/wordcount/plugin.js]: function-ize this please... what is it doing?

Comment (vpratti) [reply plugins/wordcount/plugin.js]: Checking if a related link exists first.

**Review by RyanCopley — APPROVED** [review]

File: plugins/wordcount/plugin.js [hunk 2]
```diff
@@ -201,20 +236,19 @@ CKEDITOR.plugins.add("wordcount", {
...
-
                 }
 
                 normalizedText = text;
 
                 if (!config.countSpacesAsChars) {
                     normalizedText = text.
-                        replace(/\s/g, "").
-                        replace(/&nbsp;/g, "");
+                    replace(/\s/g, "").
```
Comment (aewald) [inline plugins/wordcount/plugin.js]: why does this needs to be un-indented?

File: plugins/wordcount/plugin.js [hunk 3]
```diff
@@ -188,6 +191,38 @@ CKEDITOR.plugins.add("wordcount", {
...
+        function doSEOCheck(text, editorInstance) {
+            text = getTextBeforeFirstSEOObject(text);
+            let wordCount = countWords(text);
+            toggleSEO(editorInstance, wordCount);
+        }
+
+        function getTextBeforeFirstSEOObject(text) {
+            var assetCardSeperator = '<div asset="::asset"';
+            var relatedLinkSeperator = '<span class="exclude-from-newsgate';
+            var indexOfAssetCard = text.indexOf(assetCardSeperator);
```
Comment (RyanCopley) [inline plugins/wordcount/plugin.js:203]: Why var and not let?

File: plugins/wordcount/plugin.js [hunk 4]
```diff
@@ -188,6 +191,38 @@ CKEDITOR.plugins.add("wordcount", {
...
+            var indexOfRelatedLink = text.indexOf(relatedLinkSeperator);
+
+            //If condition to check if the relatedlink exists before a asset card.
+            if (indexOfRelatedLink != -1 && (indexOfAssetCard == -1 || indexOfRelatedLink < indexOfAssetCard)) {
+                text = text.split('<span class="exclude-from-newsgate')[0];
```
Comment (RyanCopley) [inline plugins/wordcount/plugin.js]: Why not return early?

File: plugins/wordcount/plugin.js [hunk 5]
```diff
@@ -188,6 +191,38 @@ CKEDITOR.plugins.add("wordcount", {
...
+                text = getTextBeforeFirstSEOObject('<div ' + text.slice(assetCardSeperator.length));
+            }
+
+            return text;
+        }
+
+        function toggleSEO(editorInstance, wordCount) {
+            if (wordCount < 100) {
+                document.getElementById('cke_' + editorInstance.name).classList.add('seo_warning');
+            } else if (wordCount >= 100) {
```
Comment (RyanCopley) [inline plugins/wordcount/plugin.js]: This seems silly to have an else if

File: plugins/wordcount/plugin.js (hunk 3 above)
Comment (vpratti) [reply plugins/wordcount/plugin.js:203]: To keep it consistent with the plugin file. Should we use the forked version in some project other than Presto. Again more of consistency than anything else.

**Pull request merged**